tools/src/main/java/io/github/ygojson/tools/client/ClientFactory.java

```diff
@@ -0,0 +1,67 @@
+package io.github.ygojson.tools.client;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.github.resilience4j.ratelimiter.RateLimiter;
+import io.github.resilience4j.ratelimiter.RateLimiterConfig;
+import io.github.resilience4j.ratelimiter.RateLimiterRegistry;
+import lombok.RequiredArgsConstructor;
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+import retrofit2.converter.scalars.ScalarsConverterFactory;
+
+import io.github.ygojson.tools.common.ApplicationInfo;
+
+@RequiredArgsConstructor
+public class ClientFactory {
+
+	private final ConcurrentHashMap<ClientConfig<?>, Retrofit> retrofitClients =
```

Comment:
Map should be name -> retrofit instead of using the `ClientConfig` object as a whole.